trollflow2/plugins/__init__.py

```diff
@@ -968,6 +968,11 @@ def check_valid_data_fraction(job):
     """
     logger.info("Checking valid data fraction.")
 
+    if get_twilight_poly is None:
+        logger.error("Trollsched import failed, calculation of valid data fraction not possible")
+        logger.info("Keeping all products")
+        return
```

Comment:
This should be possible to test by monkeypatching the value of `get_twilight_poly`, shouldn't it?

Reply:
I need to research what monkeypatching means.

Reply:
I was thinking [`monkeypatch.setattr`](https://docs.pytest.org/en/6.2.x/reference.html#pytest.MonkeyPatch.setattr) should do the job and take care of restoring the original value after the test is completed.

Reply:
Yep, I'm looking at what the monkeypatch means and does.

Reply:
Test added in https://github.com/pytroll/trollflow2/pull/223/commits/07023a79121ae7d9a1f6e9156a38383b99cad84d